pkg/jx/cmd/delete_application.go

```diff
@@ -312,29 +312,28 @@ func (o *DeleteApplicationOptions) waitForGitOpsPullRequest(env *v1.Environment,
 			if pr.Merged != nil && *pr.Merged {
 				log.Infof("Request %s is merged!\n", util.ColorInfo(pr.URL))
 				return nil
+			}
+			if pr.IsClosed() {
+				log.Warnf("Pull Request %s is closed\n", util.ColorInfo(pr.URL))
+				return fmt.Errorf("Promotion failed as Pull Request %s is closed without merging", pr.URL)
+			}
+			// lets try merge if the status is good
+			status, err := gitProvider.PullRequestLastCommitStatus(pr)
+			if err != nil {
+				log.Warnf("Failed to query the Pull Request last commit status for %s ref %s %s\n", pr.URL, pr.LastCommitSha, err)
 			} else {
-				if pr.IsClosed() {
-					log.Warnf("Pull Request %s is closed\n", util.ColorInfo(pr.URL))
-					return fmt.Errorf("Promotion failed as Pull Request %s is closed without merging", pr.URL)
-				}
-				// lets try merge if the status is good
-				status, err := gitProvider.PullRequestLastCommitStatus(pr)
-				if err != nil {
-					log.Warnf("Failed to query the Pull Request last commit status for %s ref %s %s\n", pr.URL, pr.LastCommitSha, err)
-				} else {
-					if status == "success" {
-						if !o.NoMergePullRequest {
-							err = gitProvider.MergePullRequest(pr, "jx promote automatically merged promotion PR")
-							if err != nil {
-								if !logMergeFailure {
-									logMergeFailure = true
-									log.Warnf("Failed to merge the Pull Request %s due to %s maybe I don't have karma?\n", pr.URL, err)
-								}
+				if status == "success" {
+					if !o.NoMergePullRequest {
+						err = gitProvider.MergePullRequest(pr, "jx promote automatically merged promotion PR")
+						if err != nil {
+							if !logMergeFailure {
```

Comment:
it looks like this boolean is the inverse of what it is controlling (if it is false then the message is logged, if true then it is not logged.

Reply:
Was a reformat